### PR TITLE
`TextFieldBase` house keeping 🧹 

### LINKED ISSUE #16184: 🧹 [UI House Keeping] TextFieldBase
### Description
Ensure that `TextFieldBase` adheres to all of the following conventions and standards

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `src`
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [x] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [x] All Base components follow the suffix convention e.g. `ButtonBase`
- [x] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [x] Add locals for any default text I18nContext as default context 
- [x] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

Updating `TextFieldBase` to include the most up to date standards and conventions for the component-library components listed in the issue and description below

* Fixes #16184 

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `src`
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [x] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [x] All Base components follow the suffix convention e.g. `ButtonBase`
- [x] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [x] Add locals for any default text I18nContext as default context 
- [x] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### Before

https://user-images.githubusercontent.com/8112138/203655191-f7412a60-7655-4e79-9352-272b125bd312.mov

### After

https://user-images.githubusercontent.com/8112138/203655303-3b8a2cc0-f0e2-4360-af6f-8b3724eae370.mov

## Manual Testing Steps

- Go to latest build of Storybook on this PR
- Search `TextFieldBase` in the search bar
- Check stories, controls and docs

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/text-field-base/text-field-base.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const TextFieldBase = ({
       setFocused(true);
     }
 
-    if (onClick) {
+    if (onClick && !disabled) {
       onClick(event);
     }
   };
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/text-field-base.js:73]: Should only fire `onClick` if not `disabled`. Have also added test for this

File: ui/components/component-library/text-field-base/text-field-base.js [hunk 2]
```diff
@@ -97,7 +97,7 @@ export const TextFieldBase = ({
         'mm-text-field-base',
         `mm-text-field-base--size-${size}`,
         {
-          'mm-text-field-base--focused': focused && !disabled && !readOnly,
+          'mm-text-field-base--focused': focused && !disabled,
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/text-field-base.js:100]: There was a suggestion to not show focused style state when an input is `readOnly` I had concerns about accessibility then and I still do so I'm removing it until someone says that it's so incredibly bad to have that I need to remove it METAMASK IS FOR THE USERS 😝

File: ui/components/component-library/text-field-base/text-field-base.stories.js [hunk 3]
```diff
@@ -10,16 +11,22 @@ import {
...
-import { Text } from '../text';
+import {
+  AvatarAccount,
+  AvatarToken,
+  Button,
+  ButtonIcon,
+  ICON_NAMES,
+  Icon,
+  Text,
+} from '..';
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/text-field-base.stories.js:22]: Shorthand imports baby!

File: ui/components/component-library/text-field-base/text-field-base.stories.js [hunk 4]
```diff
@@ -144,7 +151,13 @@ export default {
...
 };
 
-const Template = (args) => <TextFieldBase {...args} />;
+const Template = (args) => {
+  const [{ value }, updateArgs] = useArgs();
+  const handleOnChange = (e) => {
+    updateArgs({ value: e.target.value });
+  };
+  return <TextFieldBase {...args} value={value} onChange={handleOnChange} />;
+};
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/text-field-base.stories.js:160]: This allows for the value prop to work in the component and as a control


https://user-images.githubusercontent.com/8112138/203656571-bfb14167-8877-471d-96aa-e8517b783be9.mov

File: ui/components/component-library/text-field-base/text-field-base.test.js [hunk 5]
```diff
@@ -246,7 +250,7 @@ describe('TextFieldBase', () => {
     const textFieldBase = getByTestId('text-field-base');
 
     expect(textFieldBase.value).toBe(''); // initial value is empty string
-    fireEvent.change(textFieldBase, { target: { value: 'text value' } });
+    await user.type(textFieldBase, 'text value');
     expect(textFieldBase.value).toBe('text value');
     fireEvent.change(textFieldBase, { target: { value: '' } }); // reset value
     expect(textFieldBase.value).toBe(''); // value is empty string after reset
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/text-field-base.test.js:256]: Improving unit tests by using the userEvent and render helpers. Still all 🟢 

Screenshot 2022-11-23 at 2 17 28 PM

File: ui/components/component-library/text-field-base/README.mdx [hunk 6]
```diff
@@ -109,18 +107,11 @@ import { TextFieldBase } from '../../ui/components/component-library';
...
-      style={{ padding: 0 }}
-      backgroundColor={COLORS.TRANSPARENT}
-    >
-      <Icon
-        color={COLORS.PRIMARY_DEFAULT}
-        name={ICON_NAMES.SCAN_BARCODE_FILLED}
-      />
-    </Box>
+    <ButtonIcon
+      name={ICON_NAMES.SCAN_BARCODE_FILLED}
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/text-field-base/README.mdx:111]: Updated to `iconName`

File: ui/components/component-library/text-field-base/text-field-base.stories.js [hunk 7]
```diff
@@ -243,17 +256,11 @@ export const LeftAccessoryRightAccessory = (args) => {
...
-            style={{ padding: 0 }}
-            backgroundColor={COLORS.TRANSPARENT}
-          >
-            <Icon
-              color={COLORS.PRIMARY_DEFAULT}
-              name={ICON_NAMES.SCAN_BARCODE_FILLED}
-            />
-          </Box>
+          <ButtonIcon
+            name={ICON_NAMES.SCAN_BARCODE_FILLED}
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/text-field-base/text-field-base.stories.js]: Updated to `iconName`

File: ui/components/component-library/text-field-base/text-field-base.stories.js [hunk 8]
```diff
@@ -274,11 +281,11 @@ export const LeftAccessoryRightAccessory = (args) => {
             alignItems={ALIGN_ITEMS.CENTER}
           >
             <AvatarToken
-              tokenName="ast"
-              tokenImageUrl="./AST.png"
+              tokenName="eth"
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/text-field-base/text-field-base.stories.js:284]: Nice spotting. Will be fixed in `AvatarToken` house keeping https://github.com/MetaMask/metamask-extension/pull/16662

**Review by garrettbear — DISMISSED** [review]

**Review by garrettbear — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**